controller/lifecycle/lifecycle.go

```diff
@@ -46,9 +46,14 @@ type Subroutine interface {
 	Finalizers() []string
 }
 
-func NewLifecycleManager(logger *logger.Logger, operatorName string, controllerName string, client client.Client, subroutines []Subroutine) *LifecycleManager {
+func NewLifecycleManager(ctx context.Context, operatorName string, controllerName string, client client.Client, subroutines []Subroutine) *LifecycleManager {
```

Comment:
I think for functions inside of the reconcile, passing the context makes sense.
For the New functions i think we should pass the values explicitly. Since they are not operating with the context, or doing any processes that are worth canceling. 

The adding of the string parametes to the logger is great 👍